### DISCUSSION TITLE
[FEEDBACK]: Make ML06 more precise and with more Attack Scenarios

### DISCUSSION THREAD

**Discussion by mik0w** [OWASP/www-project-machine-learning-security-top-10#117] [Machine Learning Security Topics]
### Type

Suggestions for Improvement

### What would you like to report?

**Re-thinking and re-writing ML06 - corrupted packages**

The description of ML05 is quite limited given how complicated the software supply chains are, especially those related to ML-using software. 

In the summary of the vulnerability it is written: _This type of attack can be particularly dangerous as it can go unnoticed for a long time, since the victim may not realize that the package they are using has been compromised. The attacker's malicious code could be used to steal sensitive information, modify results, or even cause the machine learning model to fail._. Meanwhile, in the Detectability section in Risk Factors it says, that it's easy to detect this kind of vulns. 

What is more, there's nothing said about countermeasures such as SBOM/MLBOM etc. in the description of this vulnerability. In my opinion that should be included. 

There's plenty of resources that should be analyzed and used for the description of this specific vulnerability: 
- [CycloneDX ML-BOM](https://cyclonedx.org/capabilities/mlbom/)
- [SPDX AI](https://spdx.dev/learn/areas-of-interest/ai/)
- [this article](https://5stars217.github.io/2023-08-08-red-teaming-with-ml-models/)
- [another article](https://stiankri.substack.com/p/distribution-confusion-in-pypi)
- [or this paper](https://www.researchgate.net/publication/372761501_Assessing_the_Vulnerabilities_of_the_Open-Source_Artificial_Intelligence_AI_Landscape_A_Large-Scale_Analysis_of_the_Hugging_Face_Platform)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by shsingh** [comment]
Hi @mik0w fantastic feedback. 

Just FYI that we are in the middle of refactoring this one as per #110 as well to rename the risk type.
